drafts/taking-a-byte-of-the-carbon-cake.md

```diff
@@ -0,0 +1,291 @@
+---
+title: A Byte of the Carbon Cake
+domain: software-engineering-corner.hashnode.dev
+tags: design-patterns, software-architecture, best-practices, cloud, microservices, developers
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1730817920538/my-2iG3i9.jpeg?auto=format
+publishAs: PaulSimmmons
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+enableToc: true
+---
+
+[ // Alt cover 800x400 https://cdn.hashnode.com/res/hashnode/image/upload/v1730817687984/9TrPClX8_.jpeg?auto=format ]: #
+
+# A Byte at the Carbon Cake
+
+We all make decisions that impact the future of our planet, from our choice of food and consumer products to transport options 
+for commuting or holidays. In doing so we have become accustomed to the notion of a carbon footprint and the target of carbon neutrality.  
+
+How can we apply the same thought process to the impact of the software solutions that we produce?
+
+This article provides guidance on steps we can take today. It begins with a look at the relative cost of carbon footprints for everyday things and 
+proposes some practical steps we might adopt to make our contribution to reducing the footprint of our software, 
+something of which our customers are increasingly business-conscious.
+
+### Not in Scope
+
+Our focus is on the carbon cost of systems that run the software that we create, and we highlight just a few of many worthwhile patterns that are applicable. 
+Although we touch lightly on many topics, intentionally out of scope is the way in which cloud providers account for their carbon footprints. 
+Also note that the relative carbon footprints of everyday things are necessarily a guideline only.
+
+## What is ‘carbon cost’?
+First, let’s review the carbon cost of some popular items so that we can understand the relative cost of common software components. For this we’ll use the standard unit of Kg of carbon dioxide (KgCO2). Note that for greenhouse gases other than CO2 it is common practice to convert to CO2 as an equivalent.
```

Comment:
Out of interest: in German we abbreviate with "kg" while you use  capital-letter "Kg". I would use "kg" in an English text as well but maybe here's my chance to learn something. Do you write "Kg" with captal K in English?